hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDWriteClient.java

```diff
@@ -321,15 +324,14 @@ protected void completeCompaction(HoodieCommitMetadata metadata, JavaRDD<WriteSt
   protected JavaRDD<WriteStatus> compact(String compactionInstantTime, boolean shouldComplete) {
     HoodieSparkTable<T> table = HoodieSparkTable.create(config, context);
     preWrite(compactionInstantTime, WriteOperationType.COMPACT, table.getMetaClient());
-    HoodieTimeline pendingCompactionTimeline = table.getActiveTimeline().filterPendingCompactionTimeline();
-    HoodieInstant inflightInstant = HoodieTimeline.getCompactionInflightInstant(compactionInstantTime);
-    if (pendingCompactionTimeline.containsInstant(inflightInstant)) {
-      rollbackInflightCompaction(inflightInstant, table);
-      table.getMetaClient().reloadActiveTimeline();
-    }
     compactionTimer = metrics.getCompactionCtx();
-    HoodieWriteMetadata<JavaRDD<WriteStatus>> compactionMetadata = table.compact(context, compactionInstantTime);
-    JavaRDD<WriteStatus> statuses = compactionMetadata.getWriteStatuses();
+    HoodieSparkCopyOnWriteTable sparkCopyOnWriteTable =
+        new HoodieSparkCopyOnWriteTable(config, context, table.getMetaClient());
+    HoodieWriteMetadata<HoodieData<WriteStatus>> compactionMetadata =
+        CompactHelpers.newInstance().compact(context, table, sparkCopyOnWriteTable,
+            config, compactionInstantTime, this, new SparkTaskContextSupplier());
```

Comment:
same here, why not use existing context ?

Reply:
Right. Fixed.